citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/PersonalAccessTokenService.java

```diff
@@ -0,0 +1,63 @@
+package edu.asu.diging.citesphere.core.service.impl;
+
+import java.util.Base64;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.citesphere.core.model.oauth.IPersonalAccessToken;
+import edu.asu.diging.citesphere.core.repository.PersonalAccessTokenRepository;
+import edu.asu.diging.citesphere.core.service.IPersonalAccessTokenService;
+import edu.asu.diging.citesphere.core.user.IUserManager;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+@Service
+public class PersonalAccessTokenService implements IPersonalAccessTokenService {
+
+    private final PersonalAccessTokenRepository personalAccessTokenRepository;
+
+    private IUserManager userManager;
+
+    @Autowired
+    public PersonalAccessTokenService(PersonalAccessTokenRepository personalAccessTokenRepository,
+            IUserManager userManager) {
+        this.personalAccessTokenRepository = personalAccessTokenRepository;
+        this.userManager = userManager;
+    }
+
+    @Autowired
+    private BCryptPasswordEncoder encoder;
+
+    @Override
+    public Optional<Authentication> verifyToken(Optional<String> token) {
+        if (!token.isPresent()) {
+            return Optional.empty();
+        }
+
+        byte[] decodedTokenBytes = Base64.getDecoder().decode(token.get());
+        String decodedTokenString = new String(decodedTokenBytes);
+
+        String userName = decodedTokenString.split(":")[0];
+        String password = decodedTokenString.split(":")[1];
```

Comment:
I would first check if the string actually contains a `:`. If not, then throw an exception that tells the user that the credentials are in the wrong form/invalid.

Reply:
or return an empty optional, if that indicates that another filter might be needed